std/assembly/string.ts

```diff
@@ -80,11 +80,11 @@ import { Array } from "./array";
   }
 
   @operator("+") private static __concat(left: String, right: String): String {
-    return select<String>(left, changetype<String>("null"), left !== null).concat(right);
+    return left !== null ? left.concat(right) : right;
```

Comment:
This one could be a little tricky, depending on how the overload behaves. We may be able to remove the check here as well, but then should double check that the compiler never allows `null` values when compiling the overload. Something along the lines of

```ts
var a: string = "a";
var b: string | null = "b";
a + b // should complain
b + a // should complain
````